File: output/Llama-2-100/avatar/Java/Python/codeforces_154_A.py
def codeforces_154_A ( hi ):
    N = len ( hi )
    set = set ()
    K = int ( input () )
    res = 0
    for k in range ( K ):
        input_ = input ()
        a = 0
        b = 0
        for i in range ( N ):
            if hi [ i ] == input_ [ 0 ]:
                a += 1
            elif hi [ i ] == input_ [ 1 ]:
                b += 1
            else:
                res += min ( a, b )
                a = b = 0
        res += min ( a, b )
    print ( res )

File: output/Llama-2-100/avatar/Python/Java/codeforces_154_A.java
import java.util.Scanner;

public class codeforces_154_A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int ans = 0;
        for (int i = 0; i < n; i++) {
            String x = sc.next();
            int a = 0, b = 0;
            for (char j : x) {
                if (j == x[0]) {
                    a++;
                } else if (j == x[1]) {
                    b++;
                } else {
                    ans += Math.min(a, b);
                    a = 0;
                    b = 0;
                }
            }
            ans += Math.min(a, b);
        }
        System.out.println(ans);
    }
}


